cdist/conf/type/__rm/explorer/list

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh -e
+#
+# 2019 Ander Punnar (ander-at-kvlt-dot-ee)
+#
+# This file is part of cdist.
+#
+# cdist is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cdist is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with cdist. If not, see <http://www.gnu.org/licenses/>.
+#
+
+if [ -f "$__object/parameter/path" ]
+then
+    path="$( cat "$__object/parameter/path" )"
+else
+    path="/$__object_id"
+fi
+
+[ ! -d "$path" ] && exit 0
+
+remove="$( cat "$__object/parameter/remove" )"
+
+list="$( find "$path" -mindepth 1 -maxdepth 1 | grep -E "$remove" )"
```

Comment:
What if file contains spaces?

Why not refactor this type and use `find`'s `-path  pattern` or `-name pattern` options along with `-exec ...`?

Reply:
hmm, yes, using `find` sounds saner. check next commit.